.github/workflows/rerun-failed-run.yml

```diff
@@ -61,3 +74,17 @@ jobs:
           : # wait until the job is completed
           gh run watch ${{ inputs.run_id }} -R ${{ inputs.repository }} > /dev/null 2>&1
           gh run rerun ${{ inputs.run_id }} -R ${{ inputs.repository }} --failed
+
+      - name: Notify back error for run with id ${{ inputs.run_id }} for ${{ inputs.repository }} repository
+        if: steps.check_logs.outputs.rerun == 'false' && inputs.notify-back-on-error == 'true'
+        env:
+          GH_DEBUG: api
+          GITHUB_TOKEN: ${{ steps.github-token.outputs.token }}
+        run: |
+          : # retrieve workflow details
+          WORKFLOW_URL="${{ github.server_url }}/${{ inputs.repository }}/actions/runs/${{ inputs.run_id }}"
+          notify_back_error_message="This workflow attempted a retry but there was no match on the log message, so it was not retried. Please check: $WORKFLOW_URL"
+          WORKFLOW_NAME=$(gh run view ${{ inputs.run_id }} -R ${{ inputs.repository }} --json workflowName --jq '.workflowName')
+          echo "WORKFLOW_NAME=${WORKFLOW_NAME}"
+          : # trigger a new run with the error notify
+          gh workflow run "${WORKFLOW_NAME}.yml" -R ${{ inputs.repository }} -F notify_back_error_message="$notify_back_error_message" --ref="main"
```

Comment:
not sure about `--ref="main"`, what if the workflow has originally started not from `main` branch? (can be `master`, `stage`, `stable/8.4` etc.)